packages/compose/src/higher-order/with-state/index.js

```diff
@@ -12,11 +13,20 @@ import createHigherOrderComponent from '../../utils/create-higher-order-componen
  * A Higher Order Component used to provide and manage internal component state
  * via props.
  *
+ * @deprecated Use `useState` instead.
+ *
  * @param {?Object} initialState Optional initial state of the component.
  *
  * @return {WPComponent} Wrapped component.
  */
 export default function withState( initialState = {} ) {
+	deprecated( 'wp.compose.withState', {
+		since: '10.8',
+		plugin: 'Gutenberg',
```

Comment:
I think we should remove "since, plugin and version" from here since this API is already in Core.

Reply:
Do you mean `useState` API? Yeah, the transition shouldn't be a problem for devs. I will update the code.